File: android/model/News.kt
package com.tobiasheymann.ep.model

import java.io.Serializable

data class News(
    val metadata: NewsMetadata,
    val content: List<Content>
) : Serializable {
    companion object {
        const val COLLECTION_NAME = "news"

        fun toObject(map: Map<String, Any>): News? {
            if (map["metadata"] == null || map["content"] == null) {
                return null
            }

            val metadata = NewsMetadata.toObject(map["metadata"]!! as Map<String, Any>)
            val content = mutableListOf<Content>()
            for (item in map["content"]!! as List<Map<String, Any>>) {
                Content.toObject(item)?.let {
                    content.add(it)
                }
            }

            return if (metadata == null) {
                null
            } else {
                News(metadata!!, content)
            }
        }
    }

    val firstText: String
        get() {
            for (c in content) {
                if (c.type === ContentType.TEXT) {
                    return c.content
                }
            }
            return ""
        }
}

File: ios/model/News.swift
struct News: Serializable {
    let metadata: NewsMetadata
    let content: [Content]
    static let COLLECTION_NAME = "news"
    static func toObject(map: [String: Any]) -> News? {
        if map["metadata"] == nil || map["content"] == nil {
            return nil
        }

        let metadata = NewsMetadata.toObject(map: map["metadata"]! as! [String: Any])
        var content = [Content]()
        for item in map["content"]! as! [[String: Any]] {
            if let it = Content.toObject(map: item) {
                content.append(it)
            }
        }

        return metadata == nil ? nil : News(metadata: metadata!, content: content)
    }

    var firstText: String {
        for c in content {
            if c.type == ContentType.TEXT {
                return c.content
            }
        }

        return ""
    }
}
